pkg/kubelet/util/util.go

```diff
@@ -60,3 +66,47 @@ func GetContainerByIndex(containers []v1.Container, statuses []v1.ContainerStatu
 	}
 	return containers[idx], true
 }
+
+var (
+	tmpfsNoswapOptionSupported        bool
+	tmpfsNoswapOptionAvailabilityOnce sync.Once
+)
+
+func IsTmpfsNoswapOptionSupported(mounter mount.Interface) bool {
+	isTmpfsNoswapOptionSupportedHelper := func() bool {
```

Comment:
why define a functor? you can just put the contents of it into the Do func

Reply:
I think it's more elegant this way, as the function is very cohesive and focuses on answering the actual question of whether tmpfs noswap is supported or not. Otherwise I'd have to both identify if it's supported and assign the variable.

So it would look something like:
```go
if err != nil {
	klog.InfoS("error creating dir ...")
	tmpfsNoswapOptionSupported = false
	return
}
```

It's both unelegant and error prone IMO, for example it's easy to forget to return after the assignment is made.

But that's just one man's opinion :) If people think it's more elegant the other way I'm willing to make the change :+1:

Reply:
fair enough, this works for me :)